src/java/main/org/apache/zookeeper/ClientCnxnSocketNetty.java

```diff
@@ -340,13 +340,24 @@ public static Packet getInstance() {
             return instance;
         }
     }
+
     /**
      * ZKClientPipelineFactory is the netty pipeline factory for this netty
      * connection implementation.
      */
     private class ZKClientPipelineFactory implements ChannelPipelineFactory {
         private SSLContext sslContext = null;
         private SSLEngine sslEngine = null;
+        private String host;
+        private int port;
+
+        public ZKClientPipelineFactory() {
```

Comment:
do we still need this constructor?

Reply:
Maybe not，but  it's just a good thing to keep the default constructor when I define a constructor with parameters.

Reply:
I don't think we need it here.